### DISCUSSION TITLE
Minor error message bug

### DISCUSSION THREAD

**Discussion by brisingre** [aclap-dev/video-downloadhelper#823] [Help and Q&A]
### Your question

I love the new version! It's so much better! But I have found a couple quirks I think are worth reporting. 

1. The message saying that a video has failed to download and asking me to submit it sometimes comes up when I have a large number of videos downloading at once and they're taking a long time. As far as I can tell the videos all download fine in the end when this happens, and the message appears prematurely. Because this message doesn't say _what_ video supposedly failed, it's hard to double-check. This message would be a lot more helpful if it included the title of the video. 

2. It would significantly improve my experience if I had a button to clear all the finished downloads from the sidebar. Right now we have an option to clear them automatically (not what I want, it's useful to have them there sometimes) and we can X each one individually. I want a button that just Xs them all at once. 

Thanks for making such a useful tool. :)

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"f68231d88a2b4c9e"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 29,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

When you get that error message with the button for submitting a problem, I believe the error text includes the URL of the allegedly offending video.  You might get more information by visiting that web page.  But I think your idea of showing the title of the offending video would be better.  Best would be the full file specification of the target file on your HDD.  I say that because VDH tends to give the same title to multiple videos that might all come from a given page.  A unique identifier would instead be the target file name, including the full directory path, on your system.

For clearing all downloads without clearing everything else, that sounds like a useful idea.  I'll +1 on that.

**Comment by paulrouget** [comment]
> It would significantly improve my experience if I had a button to clear all the finished downloads from the sidebar. Right now we have an option to clear them automatically (not what I want, it's useful to have them there sometimes) and we can X each one individually. I want a button that just Xs them all at once.

Would the "Clean All" button help?

<img width="338" alt="Screenshot 2024-06-06 at 09 52 58" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/9155aab2-b3d8-4106-b8c2-01078140099a">

**Reply by GrampaWildWilly** [reply]
I think what our original poster is asking for amounts to this:

1. Clear all green ones, the ones that have been downloaded.
2. Clear all the other ones, the ones that have not been downloaded.
3. Clear everything.

We have 2 & 3.  They already exist in VDH 9.  It seems kind of like the logical next step to get 1.

**Reply by brisingre** [reply]
> I think what our original poster is asking for amounts to this:
> 
>     1. Clear all green ones, the ones that have been downloaded.
> 
>     2. Clear all the other ones, the ones that have not been downloaded.
> 
>     3. Clear everything.
> 
> 
> We have 2 & 3. They already exist in VDH 9. It seems kind of like the logical next step to get 1.

This is correct. Clean All is close, but it also removes un-downloaded stuff from the list.

**Comment by brisingre** [comment]
This got me again today! I hit Clean All to clear my downloaded files and it got rid of all my discovered media including in other tabs, I had to refresh everything. Minor inconvenience, but it would be so much better if I had a "clear downloaded" button.

I've also continued to experience the issue with getting message failed downloads when the download seems fine. And I can't double-check to be sure, because I never know what file supposedly failed, just that something did...

**Comment by ThomasBecker42** [comment]
I have the same problem as mentioned. 
When downloading multiple videos and an "Download failed" I have no chance to see which one it was.
With the old UI/Version I was able to see the name of the video.
With ths version the error message is completely generic, no URL and no Title. 
Firefox 127.0.2 (64-Bit)
Version 9.0.2.12 (stable), target mozilla, locale de.

**Comment by GrampaWildWilly** [comment]
Se my response in #948.
